### PR TITLE
convert split into a standalone optional category, implement add noise function inside CsvProcessing, and add protein experiment

### PR DESCRIPTION
**Pull request by suzannejin**
relate to issue #32 and #21

### CODE REVIEW

**Review by mathysgrapotte — DISMISSED** [review]
See comments

File: bin/src/data/experiments.py [hunk 1]
```diff
@@ -32,45 +30,34 @@ def get_split_indexes(self, data: list, split: tuple) -> list | list | list:
...
-            else:
-                return [key for key in data if data_type in key.split(':')[1]]
         
     def get_encoding_all(self, data_type: str) -> Any:
         """
         This method gets the encoding function for a specific data type.
         """
         return getattr(self, data_type)['encoder'].encode_all
     
+    def add_noise_all(self, data_type: str, noise_generator: str) -> Any:
```
Comment (alessiovignoli) [inline bin/src/data/experiments.py:40]: rename this fgunction to noise_key ? or noise_all ?  so to not habve the same name in the below function class

File: bin/src/data/csv.py [hunk 2]
```diff
@@ -17,48 +17,154 @@
...
+        """
+        Adds noise to the data.
+        Noise is added for each column with the configurations specified in the configs list.
+        """
+        for dictionary in configs:
+            key = dictionary['column_name']
+            data_type = key.split(':')[2]
+            noise_generator = dictionary['name']
+            new_column = self.experiment.add_noise_all(data_type, noise_generator)(list(self.data[key]), **dictionary['params'])
+            self.data = self.data.with_columns(pl.Series(key, new_column))
```
Comment (alessiovignoli) [inline bin/src/data/csv.py:92]: add a comment explaining this line and the above plese for better readability

File: bin/src/data/csv.py [hunk 3]
```diff
@@ -17,48 +17,154 @@
...
+        Saves the data to a csv file.
+        """
+        self.data.write_csv(path)
+
+    
+class CsvLoader(CsvHandler):
     """
+    Class for loading and splitting the csv data, and then encode the information.
     
+    It will parse the CSV file into four dictionaries, one for each category [input, label, split].
```
Comment (alessiovignoli) [inline bin/src/data/csv.py:105]: add meta as category

File: bin/src/data/csv.py [hunk 4]
```diff
@@ -102,56 +208,17 @@ def __len__(self) -> int:
...
-
-    def noise(self, data):
-        """
-        Adds noise to the data.
-        """
-        pass
+        meta = {}
+        for key in self.meta:
+            data = self.meta[key][idx]
+            if not isinstance(data, list):
```
Comment (alessiovignoli) [inline bin/src/data/csv.py:224]: add comment to explain this if

File: bin/src/data/csv.py [hunk 5]
```diff
@@ -17,48 +17,154 @@
...
-        """
-        Loads the csv file into a polars dataframe.
-        """
-        return pl.read_csv(csv_path)
+    def load_csv_per_split(self, split: int) -> pl.DataFrame:
+        """
+        Load the part of csv file that has the specified split value.
+        Split is a number that for 0 is train, 1 is validation, 2 is test.
+        This is accessed through the column with category `split`. Example column name could be `split:split:int`.
+        """
```
Comment (alessiovignoli) [inline bin/src/data/csv.py:139]: please say that this funcxtion is thought to be called in the training step.

**Review by alessiovignoli — APPROVED** [review]

**Pull request merged**